### ISSUE TITLE
Urgent problem of partial modify and provide delete function

### ISSUE THREAD

**Issue by MichaelPanwenjie** [josdejong/jsoneditor#1463]
I am using jsonedit to add, delete, check and change mongodb data in the angular project. 
What I want now is ：
1. Let jsonedit only can modify value, can not modify keys,
2. You cannot add data structures. You can delete some structures, but you cannot add additional ones
3. Jsonedit has the function of remove
4. Verify data validity

How do you realize this requirement？can give me a simple, thanks!

**Comment by josdejong** [comment]
Have you see the options `onEditable`, `schema`, and `onValidate`? I think these options can more or less cover your list.

See docs: https://github.com/josdejong/jsoneditor/blob/master/docs/api.md
And you can search the examples for usage examples of the mentioned options: https://github.com/josdejong/jsoneditor/tree/develop/examples

**Comment by MichaelPanwenjie** [comment]
I used onEditable function, it can make keys uneditable,but when i using it the remove function has been lost.

Can you give me some short examples or related links to reference ,it is urgent.  Thanks a lot!

**Comment by josdejong** [comment]
> I used onEditable function, it can make keys uneditable,but when i using it the remove function has been lost.

I'm not sure what you mean exactly. When you make a key uneditable, you can indeed not remove this field.

**Comment by MichaelPanwenjie** [comment]
Okay, I see.  - But can I control the insert function? If all the keys exist then disable insert, If any key is missed then enable insert function?  Waiting for your replay,thanks!

**Comment by josdejong** [comment]
Yes, you can modify the buttons shown in the context menu with the option `onCreateMenu` if the default behavior of what `onEditable` disabled does not match your needs.

**Comment by MichaelPanwenjie** [comment]
but how can  i  get the context menus ?  From any public object ?

**Comment by josdejong** [comment]
No, you can implement the `onCreateMenu` callback and in that callback you can add or remove buttons. There is an example here: https://github.com/josdejong/jsoneditor/blob/develop/examples/21_customize_context_menu.html
